src/codegen/codegen_driver.cpp

```diff
@@ -0,0 +1,262 @@
+/*************************************************************************
+ * Copyright (C) 2018-2022 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#include <string>
+
+#include "codegen/codegen_driver.hpp"
+#include "codegen_compatibility_visitor.hpp"
+#include "utils/logger.hpp"
+#include "visitors/ast_visitor.hpp"
+#include "visitors/semantic_analysis_visitor.hpp"
+#include "visitors/symtab_visitor.hpp"
+#include "visitors/after_cvode_to_cnexp_visitor.hpp"
+#include "visitors/perf_visitor.hpp"
+#include "visitors/global_var_visitor.hpp"
+#include "visitors/local_to_assigned_visitor.hpp"
+#include "visitors/verbatim_var_rename_visitor.hpp"
+#include "visitors/constant_folder_visitor.hpp"
+#include "visitors/loop_unroll_visitor.hpp"
+#include "visitors/kinetic_block_visitor.hpp"
+#include "visitors/steadystate_visitor.hpp"
+#include "visitors/inline_visitor.hpp"
+#include "visitors/local_var_rename_visitor.hpp"
+#include "visitors/localize_visitor.hpp"
+#include "visitors/sympy_conductance_visitor.hpp"
+#include "visitors/sympy_solver_visitor.hpp"
+#include "visitors/neuron_solve_visitor.hpp"
+#include "visitors/solve_block_visitor.hpp"
+#include "visitors/nmodl_visitor.hpp"
+#include "visitors/units_visitor.hpp"
+#include "visitors/ispc_rename_visitor.hpp"
+
+using namespace nmodl;
+using namespace codegen;
+using namespace visitor;
+
+bool CodegenDriver::prepare_mod(ast::Program& node) {
+    /// whether to update existing symbol table or create new
+    /// one whenever we run symtab visitor.
+    bool update_symtab = false;
+
+    std::string modfile;
+    std::string scratch_dir = "tmp";
+    auto filepath = [scratch_dir, modfile](const std::string& suffix, const std::string& ext) {
+        static int count = 0;
+        return "{}/{}.{}.{}.{}"_format(
+            scratch_dir, modfile, std::to_string(count++), suffix, ext);
+    };
+
+    /// just visit the ast
+    AstVisitor().visit_program(node);
+
+    /// Check some rules that ast should follow
+    {
+        logger->info("Running semantic analysis visitor");
+        if (nmodl::visitor::SemanticAnalysisVisitor().check(node)) {
+            return false;
+        }
+    }
+
+    /// construct symbol table
+    {
+        logger->info("Running symtab visitor");
+        nmodl::visitor::SymtabVisitor(update_symtab).visit_program(node);
```

Comment:
```suggestion
        SymtabVisitor(update_symtab).visit_program(node);
```
This could be like this since there is
```
using namespace nmodl;
using namespace codegen;
using namespace visitor;
```
above?
I don't think it should be related to the error though..
Similar thing can be applied for the visitors called above here

Reply:
It's not related but I noticed this too, cleaned it up. Thanks!